app/controllers/sessions_controller.rb

```diff
@@ -1,6 +1,38 @@
 class SessionsController < ApplicationController 
 
-    def index
-        
+    def new
+        if !current_user.nil?
+            if current_admin?
+                redirect_to admin_path
+            elsif current_merchant?
+                redirect_to merchant_path
+            else current_regular?
+                redirect_to profile_path
+            end
+        flash[:notice] = "You are already logged in!"
+        end
     end
-end
\ No newline at end of file
+
+    def create
```

Comment:
I like how this logic makes it so we don't need to create a welcome method. Also I like the idea of have a helper method for the routing. Lets look into that soon